### ISSUE TITLE
applet-window-buttons: update for KDE 6

### ISSUE THREAD

**Issue by dr460nf1r3** [NixOS/nixpkgs#294478]
### Describe the bug
The current version of `libsForQt5.applet-window-buttons` is made for Qt5 and needs to be updated for usage with KDE 6 since it does not work anymore.

### Steps To Reproduce
1. Have applet active
2. Observe an error message that its not made for KDE 6

### Expected behavior
The applet works.

### Additional context
- There is already a [PR to the main repo](https://github.com/psifidotos/applet-window-buttons/pull/214) that fixes the application, Arch uses it to [provide this applet](https://gitlab.archlinux.org/archlinux/packaging/packages/plasma-applet-window-buttons/-/commit/bd7034be7455eb5c979e1e27a668b9db86b8109e) for example. I'm using this myself and it works fairly well.
- Given the maintainer was inactive for a very long time, it is almost safe to assume that this PR won't be merged.
- The patch can be found [here](https://patch-diff.githubusercontent.com/raw/psifidotos/applet-window-buttons/pull/214.patch). 

### Notify maintainers

@dotlambda 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
/nix/store/w3cx1ir83mgri4ypsix0121v0mykhqk1-nix-info
copying path '/nix/store/w3cx1ir83mgri4ypsix0121v0mykhqk1-nix-info' from 'https://cache.garnix.io'...
- system: `"x86_64-linux"`
- host os: `Linux 6.7.8-2-cachyos, Garuda Linux, noversion, rolling`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.19.3`
- channels(root): `"home-manager, nixos"`
- nixpkgs: `/nix/store/8ny3r3jvj17ljzc89h9rpfr56a7nyn8c-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
Excuse me?

**Comment by dr460nf1r3** [comment]
Oh god, you are right 🫠 that's absolutely my bad. Sorry again.

**Comment by Leptopoda** [comment]
I was able to get it running with:
```nix
{ lib, stdenv, fetchgit, cmake, extra-cmake-modules, kcolorscheme, kconfig
, kcoreaddons, kdecoration, ki18n, kirigami, kitemmodels, ksvg, libplasma
, plasma-workspace, qt6 }:

stdenv.mkDerivation rec {
  pname = "applet-window-buttons6";
  version = "0.13.0";

  src = fetchgit {
    #owner = "moodyhunter";
    #repo = "applet-window-buttons";
    url = "https://github.com/moodyhunter/applet-window-buttons6.git";
    rev = "326382805641d340c9902689b549e4488682f553";
    hash = "sha256-POr56g3zqs10tmCbKN+QcF6P6OL84tQNkA+Jtk1LUfY=";
  };

  nativeBuildInputs = [ cmake extra-cmake-modules qt6.wrapQtAppsHook ];

  buildInputs = [
    kcolorscheme
    kconfig
    kcoreaddons
    kdecoration
    ki18n
    kirigami
    kitemmodels
    ksvg
    libplasma
    plasma-workspace
    qt6.qtbase
    qt6.qtdeclarative
  ];

  meta = with lib; {
    description =
      "Plasma 6 applet in order to show window buttons in your panels";
    homepage = "https://github.com/psifidotos/applet-window-buttons";
    license = licenses.gpl2Plus;
    maintainers = with maintainers; [ dotlambda ];
  };
}
```

Which can later be included like any other package `(kdePackages.callPackage ./applet-window-buttons.nix { })`.
I'm new to nix and not sure if I did everything right (I also had issues using fetchGitHub) but I'll try to upsream this in the coming days.

**Comment by BBFifield** [comment]
> I was able to get it running with:
>

> Which can later be included like any other package `(kdePackages.callPackage ./applet-window-buttons.nix { })`. I'm new to nix and not sure if I did everything right (I also had issues using fetchGitHub) but I'll try to upsream this in the coming days.

Good stuff! I'm also quite new to nix and appreciate the effort you went through to make this module. It works for me with the exception of it always defaulting to "Always Visible" even when "visibility" is set to anything else in Plasma-Manager with overrideConfig enabled. Likely an issue with plasma-manager, but I'm not 100% sure.

**Comment by dotlambda** [comment]
Closing because it's unacceptable you assign others who aren't working on the issue.

**Comment by Leptopoda** [comment]
You are right that it was not ok from op to assign anyone but the issue is still there.
Can someone please reopen it?

**Comment by Leptopoda** [comment]
> > I was able to get it running with:
> >
> 
> > Which can later be included like any other package `(kdePackages.callPackage ./applet-window-buttons.nix { })`. I'm new to nix and not sure if I did everything right (I also had issues using fetchGitHub) but I'll try to upsream this in the coming days.
> 
> Good stuff! I'm also quite new to nix and appreciate the effort you went through to make this module. It works for me with the exception of it always defaulting to "Always Visible" even when "visibility" is set to anything else in Plasma-Manager with overrideConfig enabled. Likely an issue with plasma-manager, but I'm not 100% sure.
> 

I too have this issue and hop that I'll get it fixed soon. Once I have more time and everythkng is working ok I'll make a PR to get it fixed upstream

**Comment by ShalokShalom** [comment]
@dotlambda The [search](https://search.nixos.org/packages?channel=24.05&show=libsForQt5.applet-window-buttons&from=0&size=50&sort=relevance&type=packages&query=applet-window-buttons) lists you as the maintainer. Does this mean, this is not up to date anymore?

**Comment by dotlambda** [comment]
> @dotlambda The [search](https://search.nixos.org/packages?channel=24.05&show=libsForQt5.applet-window-buttons&from=0&size=50&sort=relevance&type=packages&query=applet-window-buttons) lists you as the maintainer. Does this mean, this is not up to date anymore?

It is but that doesn't mean you can expect me to fix every issue with the package. Besides, someone being assigned means they are working on it which I'm not.

**Comment by ShalokShalom** [comment]
No expectations. ☺️

**Comment by Leptopoda** [comment]
@dotlambda can you please re open this issue so it is still tracked?

**Comment by Leptopoda** [comment]
Ok it looks like a new package built directly from the fork has arised:
https://github.com/NixOS/nixpkgs/blob/6212d8033f4fa94ac48fd5befca1403babaf1b06/pkgs/kde/third-party/applet-window-buttons6/default.nix

Maybe the old package should just be removed?
I have no idea of the process so a friendly ping to the new and old maintainer @A1ca7raz and @dotlambda so we can find a solution.

**Comment by A1ca7raz** [comment]
> Maybe the old package should just be removed?

Unless KDE5 is removed completely from nixpkgs, I think we should keep the old package.

**Comment by FlorianFranzen** [comment]
I will close this issue again. The request package was added in #328324 and can still be now found under [`kdePackages.applet-window-buttons6`](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=applet-window-buttons6).
